### PR TITLE
Max out smarty grumpiness in unit tests - obviously failed - need to figure out why (tests should have failed)

### PR DESCRIPTION
**Pull request by eileenmcnaughton**
Overview
----------------------------------------
Max out smarty grumpiness in unit tests

Before
----------------------------------------
Happy smarty

After
----------------------------------------
Frowny smarty

Technical Details
----------------------------------------

Comments
----------------------------------------

### CODE REVIEW

File: CRM/Core/Smarty.php [hunk 1]
```diff
@@ -158,6 +158,17 @@ private function initialize() {
...
 
+  /**
+   * Is this a test environment.
+   *
+   * Currently we just check the class name but it is possible to run tests in CMS frameworks.
+   *
+   * @return bool
+   */
+  private static function isTestEnvironment(): bool {
+    return CRM_Core_Config::singleton()->userFramework === 'CRM_Utils_System_UnitTests';
```
Comment (demeritcowboy) [inline CRM/Core/Smarty.php:169]: Is this right? Would it just be 'UnitTests'?

Comment (demeritcowboy) [reply CRM/Core/Smarty.php:169]: Also bootstrap [defines CIVICRM_TEST](https://github.com/civicrm/civicrm-core/blob/8104b62cb9b474aae6520139b02dd98d2eb9dc53/tests/phpunit/CiviTest/bootstrap.php#L17) and I've seen that used elsewhere.

**Comment by demeritcowboy** [comment]
Just looking at the log so far this broke jenkins while still at the letter "A", so most of the `CRM_` tests didn't run.

👍

**Comment by eileenmcnaughton** [comment]
Looks like it started failing here

not ok 224 - Error: CRM_Case_BAO_CaseTest::testCaseActivity
not ok 225 - Error: CRM_Case_BAO_CaseTest::testAddCaseToContact
not ok 226 - Error: CRM_Case_BAO_CaseTest::testSortByCaseContact

**Comment by demeritcowboy** [comment]
I see `not ok 165 - Error: CRM_Activity_Form_ActivityTest::testActivityCreate`

**Comment by eileenmcnaughton** [comment]
UG - that one passed for me locally - even running from the start!

There is one known test failing thing open - #22470 is what is blocking tests in https://github.com/civicrm/civicrm-core/pull/22466  - not sure if that could also affect  here

**Comment by demeritcowboy** [comment]
Ok I can look at 22470.

**Comment by eileenmcnaughton** [comment]
I've found the thing in the activity tests (or at least one thing)

**Comment by eileenmcnaughton** [comment]
This looks like the current fail -

image


I think #22470 will get past that notice

**Comment by eileenmcnaughton** [comment]
currently falling over on the isset on totalTaxAmount in the contribution template - this value could be cast from the `{contribution.tax_amount}` token - but needs to be assigned in an unformatted style #22547 would be the pre-requisite to fixing

**Comment by eileenmcnaughton** [comment]
ok - this would fix the current fail https://github.com/civicrm/civicrm-core/pull/22560

**Pull request closed without merging**